bedrock/firefox/templates/firefox/releases/release-notes.html

```diff
@@ -314,6 +314,7 @@
       <a rel="external" href="https://blog.mozilla.org/press/kits/">{{ _('Press Kit') }}</a>
       <a rel="external" href="https://blog.mozilla.org/press/">{{ _('Mozilla and Firefox News') }}</a>
       <a href="{{ url('firefox.organizations.organizations') }}">{{ _('Firefox Extended Support Release') }}</a>
+      <a href="{{ url('firefox.releases.index') }}">{{ _('All Firefox Releases') }}</a>
```

Comment:
Note: we should probably stick to title case for consistency with the other links here `All Firefox for Desktop Releases`